src/main/java/Application/OpenCVClient/OpenCVClientOperator.java

```diff
@@ -0,0 +1,269 @@
+package Application.OpenCVClient;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+import java.net.*;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+
+import Application.Utilities.DetectMarker;
+import Application.Utilities.HeartBeatRunner;
+import Application.Utilities.ImageProcessor;
+import Application.Utilities.OpenCVUtil;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import org.opencv.core.Mat;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import java.util.concurrent.Semaphore;
+
+public class OpenCVClientOperator {
+    private Socket clientSocket;
+    private BufferedReader in;
+    private DataOutputStream out;
+    private DetectMarker detector;
+    private ObjectMapper mapper;
+
+    private String fileName = "singlemarkerssource.png";
+    private String resultFileName = "detected.png";
+    private String fileExtension = "png";
+    private Mat subject;
+    private BufferedImage currentImage;
+
+    public Semaphore ipLock;
+    public Semaphore serverLock;
+
+    //Server Communication Properties
+    private int utilizeServer;
```

Comment:
it was initially a parameter received as the main argument, so I used int for more convenience
but since we've updated other input methods where this convenience is no longer relevant, I agree with you that boolean could work better